demos/common/src/ie_config_helper.cpp

```diff
@@ -0,0 +1,95 @@
+    }
+}
+
+std::map<std::string, std::string> createConfig(const std::string& deviceString,
+                                                const std::string& nstreamsString,
+                                                int nthreads,
+                                                bool minLatency) {
+    std::map<std::string, std::string> config;
+
+    std::set<std::string> devices;
+    for (const std::string& device : parseDevices(deviceString)) {
+        devices.insert(device);
+    }
+    std::map<std::string, unsigned> deviceNstreams = parseValuePerDevice(devices, nstreamsString);
+    
+    for (auto& device : devices) {
+        if (device == "CPU") {  // CPU supports a few special performance-oriented keys
+            if (minLatency) {
+                config.insert({ CONFIG_KEY(CPU_THROUGHPUT_STREAMS), "1" });
+                continue;
+            }
+            
+            // limit threading for CPU portion of inference
+            if (nthreads != 0)
+                config.insert({ CONFIG_KEY(CPU_THREADS_NUM), std::to_string(nthreads) });
+
+            if (deviceString.find("MULTI") != std::string::npos && devices.find("GPU") != devices.end()) {
+                config.insert({ CONFIG_KEY(CPU_BIND_THREAD), CONFIG_VALUE(NO) });
+            } else {
+                // pin threads for CPU portion of inference
+                config.insert({ CONFIG_KEY(CPU_BIND_THREAD), CONFIG_VALUE(YES) });
+            }
+
+            // for CPU execution, more throughput-oriented execution via streams
+            config.insert({ CONFIG_KEY(CPU_THROUGHPUT_STREAMS), (deviceNstreams.count(device) > 0
+                                                                 ? std::to_string(deviceNstreams.at(device))
+                                                                 : CONFIG_VALUE(CPU_THROUGHPUT_AUTO)) });
+        } else if (device == "GPU") {
+            if (minLatency) {
+                config.insert({ CONFIG_KEY(GPU_THROUGHPUT_STREAMS), "1" });
+                continue;
+            }
+
+            config.insert({ CONFIG_KEY(GPU_THROUGHPUT_STREAMS), (deviceNstreams.count(device) > 0
+                                                                 ? std::to_string(deviceNstreams.at(device))
+                                                                 : CONFIG_VALUE(GPU_THROUGHPUT_AUTO)) });
+
+            if (deviceString.find("MULTI") != std::string::npos && devices.find("CPU") != devices.end()) {
+                // multi-device execution with the CPU + GPU performs best with GPU throttling hint,
+                // which releases another CPU thread (that is otherwise used by the GPU driver for active polling)
+                config.insert({ CLDNN_CONFIG_KEY(PLUGIN_THROTTLE), "1" });
+            }
+        } else if (device == "MYRIAD") {
+            if (!nstreamsString.empty()) {
+                config.insert({ CONFIG_KEY(CPU_THROUGHPUT_STREAMS), "1" });
```

Comment:
I think this is the option you need https://github.com/openvinotoolkit/openvino/blob/f17262e05f448bc69b2162c137ce0736d3e48cfb/inference-engine/include/vpu/myriad_config.hpp#L60

You have
> Force Myriad to use 1 stream in demos which don't support multiple streams

in the description. But apply the config to the one that does support multiple streams.

There could be two different functions:
* the smart one for classification demo and object detection ssd demo
* naive for simple demos using only one stream

Reply:
I've tried this option `VPU_MYRIAD_CONFIG_KEY(MYRIAD_THROUGHPUT_STREAMS)`, but the build fails because `MYRIAD_THROUGHPUT_STREAMS` was not declared.

I use recent Openvino version (built from Github repo yesterday) and include `vpu/myriad_config.hpp`. There is an assumption that this thing was not fully implemented yet (the commit https://github.com/openvinotoolkit/openvino/commit/66aa119b11ba69977b37d25872951a9e17a3565c was merged only a few days ago). Maybe I should open an issue.

Reply:
> `MYRIAD_THROUGHPUT_STREAMS` was not declared

How is this possible its right in `vpu/myriad_config.hpp`.

According to the commit you pointed `ie::MYRIAD_THROUGHPUT_STREAMS` is supposed to be used directly instead of `VPU_MYRIAD_CONFIG_KEY(THROUGHPUT_STREAMS)` (or `VPU_MYRIAD_CONFIG_KEY(MYRIAD_THROUGHPUT_STREAMS)` which I can't find to be mentioned in the diff at all). But it's better to ask the author.